### PR TITLE
Debug endpoints

### PR DESCRIPTION
**Pull request by davidfurey**
Adds some useful endpoints for debug/qa purposes:

`/udids/:udid` - lookup registrations by unique device identifier
`/topics/*topic` - lookup registrations by topic
`/registrations/:platform/:deviceToken` - lookup registrations by notifications platform specific token

Feedback particularly welcome on the route naming.

### CODE REVIEW

**Comment by DiegoVazquezNanini** [comment]
:+1: LGTM.
An alternative for /udids/:udid could be /device/:id or /udid/:udid

File: registration/app/registration/services/azure/NotificationsHubRegistrar.scala [hunk 1]
```diff
@@ -5,19 +5,21 @@ import azure._
...
 import tracking.{SubscriptionTracker, TopicSubscriptionTracking}
 
 import scala.concurrent.{ExecutionContext, Future}
 import cats.data.{Xor, XorT}
 import cats.implicits._
+import models.pagination.{Paginated, ProviderCursor}
 
 class NotificationHubRegistrar(
-  hubClient: NotificationHubClient,
+  val hubClient: NotificationHubClient,
```
Comment (DiegoVazquezNanini) [inline registration/app/registration/services/azure/NotificationsHubRegistrar.scala]: similar comment than before

File: registration/app/registration/controllers/Main.scala [hunk 2]
```diff
@@ -75,6 +77,46 @@ final class Main(
...
+      registrations <- XorT(registrar.findRegistrations(deviceToken): Future[Xor[NotificationsError, List[StoredRegistration]]])
+    } yield registrations
+    result.fold(processErrors, res => Ok(Json.toJson(res)))
+  }
+
+  def registrationsByUdid(udid: UniqueDeviceIdentifier): Action[AnyContent] = Action.async {
+    val results = registrarProvider.withAllRegistrars { registrar =>
+      registrar.findRegistrations(udid)
+    }
+    Future.sequence(results).map(_.flatMap(_.toList.flatMap(_.results))).map(res => Ok(Json.toJson(res)))
```
Comment (DiegoVazquezNanini) [inline registration/app/registration/controllers/Main.scala]: Maybe you are doing too much on a single line. What do you think about creating an function.

**Comment by davidfurey** [comment]
@DiegoVazquezNanini I've changed the routes and probably a few other things along the way.  Can you look again please?

File: registration/app/registration/services/NotificationRegistrar.scala [hunk 3]
```diff
@@ -14,8 +15,31 @@ object RegistrationResponse {
...
+  import play.api.libs.json._
+
+  implicit val jf = Json.format[StoredRegistration]
+
+  def fromRegistration(registration: Registration) = {
+    StoredRegistration(
+      deviceId = registration.deviceId,
+      platform = registration.platform,
+      userId = Some(registration.udid),
+      tagIds = registration.topics.map(_.id),
```
Comment (DiegoVazquezNanini) [inline registration/app/registration/services/NotificationRegistrar.scala]: Do you think we need tagIds and topics? We can always go from the list topics to the list of topics Id.

**Comment by DiegoVazquezNanini** [comment]
:+1: LGTM

File: registration/app/registration/services/NotificationRegistrar.scala (hunk 3 above)
Comment (DiegoVazquezNanini) [inline registration/app/registration/services/NotificationRegistrar.scala]: Ok then

**Pull request merged**